dev/devicelab/lib/tasks/run_without_leak.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../framework/adb.dart';
+import '../framework/framework.dart';
+import '../framework/utils.dart';
+
+TaskFunction createRunWithoutLeakTest(dynamic dir) {
+  return () async {
+    final Device device = await devices.workingDevice;
+    await device.unlock();
+    final List<String> options = <String>[
+      '-d', device.deviceId, '--verbose',
+    ];
+    setLocalEngineOptionIfNecessary(options);
+    int exitCode;
+    await inDirectory<void>(dir, () async {
+      final Process process = await startProcess(
+          path.join(flutterDirectory.path, 'bin', 'flutter'),
+          <String>['run']..addAll(options),
+          environment: null,
+      );
+      final Completer<void> stdoutDone = Completer<void>();
+      final Completer<void> stderrDone = Completer<void>();
+      process.stdout
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        if (line.contains('\] For a more detailed help message, press "h". To detach, press "d"; to quit, press "q"')) {
+          process.stdin.writeln('q');
+        }
+        print('stdout: $line');
+      }, onDone: () {
+        stdoutDone.complete();
+      });
+      process.stderr
+          .transform<String>(utf8.decoder)
+          .transform<String>(const LineSplitter())
+          .listen((String line) {
+        print('stderr: $line');
+      }, onDone: () {
+        stderrDone.complete();
+      });
+
+      await Future.wait<void>(
```

Comment:
Is there any sort of timeout process in the devicelab? I worry that if we change  "] For a more detailed help message, press "h". To detach, press "d"; to quit, press "q"'" we're gonna get some odd failures.

Reply:
15 minutes.

I'm not sure of a better line to look for - I'm open to suggestions, but we could also just catch this failure and update it worst case.

Reply:
As long as it times out eventually this is good for now